src/app.rs

```diff
@@ -512,8 +641,16 @@ impl Entry for Slot {
         cx.slot_rect = Some(rect); // Save slot rect for use later
 
         let response = ui.allocate_rect(rect, egui::Sense::hover());
+
+        let clicked = ui.input().pointer.any_click()
+            && rect.contains(ui.input().pointer.interact_pos().unwrap());
+
         let mut hover_pos = response.hover_pos(); // where is the mouse hovering?
 
+        if cx.selected_node.is_some() && cx.selected_node.clone().unwrap().entry_id == self.entry_id
```

Comment:
It should be possible to do this without cloning, e.g., using `ref`:

```rust
if let Some(ref selected) = cx.selected_node {
    if selected.entry_id == self.entry_id {
       ...
    }
}
```

(I may not have the syntax exactly right, you may need to move the `ref` around.)

Requires doubling up the `if` statement, but clearer overall.